File: frontend/src/types/stock.ts
export interface StockInfo {
  symbol: string
  name: string
  pe_ratio?: number
  pb_ratio?: number
  dividend_rate?: number
  debt_ratio?: number
  market_cap?: number
  industry?: string
  range_change?: number
  main_fund_inflow?: number
}

export interface StockListResponse {
  success: boolean
  message: string
  count: number
  stocks: StockInfo[]
  timestamp: string
}

export interface ValueStockRequest {
  top_n: number
  pe_max?: number
  pb_max?: number
}

export interface MainForceStockRequest {
  days_ago: number
  min_market_cap: number
  max_market_cap: number
  max_range_change: number
  top_n: number
}

export interface TechnicalIndicator {
  name: string
  value?: number
  signal?: string
  description?: string
}

export interface KlineData {
  date: string
  open: number
  close: number
  high: number
  low: number
  volume: number
  ma5?: number
  ma10?: number
  ma20?: number
}

export interface TechnicalAnalysisRequest {
  symbol: string
  days_ago: number
}

export interface BatchAIAnalysisRequest {
  symbols: string[]
  days_ago?: number
}

export interface BatchAIAnalysisResponse {
  success: boolean
  message: string
  total: number
  success_count: number
  failed_count: number
  results: AIAnalysisResponse[]
  failed_symbols: string[]
}

export interface TechnicalAnalysisResponse {
  success: boolean
  message: string
  symbol: string
  name?: string
  current_price?: number
  change_percent?: number
  indicators: TechnicalIndicator[]
  kline_data?: KlineData[]
  summary?: string
  timestamp: string
}

export interface FundamentalMetric {
  name: string
  value?: number
  unit?: string
  rank?: string
  description?: string
}

export interface FundamentalAnalysisRequest {
  symbol: string
}

export interface FundamentalAnalysisResponse {
  success: boolean
  message: string
  symbol: string
  name?: string
  industry?: string
  market_cap?: number
  pe_ratio?: number
  pb_ratio?: number
  roe?: number
  debt_ratio?: number
  dividend_rate?: number
  metrics: FundamentalMetric[]
  summary?: string
  timestamp: string
}

export interface AnalysisScore {
  category: string
  score: number
  weight: number
}

export interface AIAnalysisRequest {
  symbol: string
}

export interface AIAnalysisResponse {
  success: boolean
  message: string
  symbol: string
  name?: string
  ai_report: string
  timestamp: string
}

export interface ComprehensiveAnalysisRequest {
  symbol: string
  days_ago: number
}

export interface ComprehensiveAnalysisResponse {
  success: boolean
  message: string
  symbol: string
  name?: string
  overall_score: number
  overall_rating: string
  scores: AnalysisScore[]
  technical_analysis?: TechnicalAnalysisResponse
  fundamental_analysis?: FundamentalAnalysisResponse
  risks: string[]
  opportunities: string[]
  recommendations: string[]
  timestamp: string
}

export interface AnalysisHistoryItem {
  id: string
  symbol: string
  name?: string
  created_at: string
  ai_report?: string
}

export interface AnalysisHistoryListResponse {
  success: boolean
  message: string
  total: number
  items: AnalysisHistoryItem[]
}

export interface AnalysisHistoryDetailResponse {
  success: boolean
  message: string
  data?: AnalysisHistoryItem
}

File: backend/schemas/stock.py
from pydantic import BaseModel, Field
from typing import Optional, List, Dict, Any
from datetime import datetime


class ValueStockRequest(BaseModel):
    pe_max: float = Field(default=20.0, description="最大市盈率")
    pb_max: float = Field(default=1.5, description="最大市净率")
    top_n: int = Field(default=10, description="返回前N只股票", ge=5, le=50)


class MainForceStockRequest(BaseModel):
    days_ago: int = Field(default=1, description="距今多少天", ge=1, le=30)
    min_market_cap: float = Field(default=50.0, description="最小市值（亿）", ge=1.0)
    max_market_cap: float = Field(default=500.0, description="最大市值（亿）", ge=1.0)
    max_range_change: float = Field(default=20.0, description="最大涨跌幅限制（%）", ge=1.0)
    top_n: int = Field(default=20, description="返回前N只股票", ge=5, le=100)


class StockInfo(BaseModel):
    symbol: str = Field(..., description="股票代码")
    name: str = Field(..., description="股票名称")
    pe_ratio: Optional[float] = Field(None, description="市盈率")
    pb_ratio: Optional[float] = Field(None, description="市净率")
    dividend_rate: Optional[float] = Field(None, description="股息率")
    debt_ratio: Optional[float] = Field(None, description="资产负债率")
    market_cap: Optional[float] = Field(None, description="市值")
    industry: Optional[str] = Field(None, description="所属行业")
    range_change: Optional[float] = Field(None, description="区间涨跌幅")
    main_fund_inflow: Optional[float] = Field(None, description="主力资金净流入")


class StockListResponse(BaseModel):
    success: bool = Field(..., description="是否成功")
    message: str = Field(..., description="消息")
    count: int = Field(default=0, description="股票数量")
    stocks: List[StockInfo] = Field(default_factory=list, description="股票列表")
    timestamp: datetime = Field(default_factory=datetime.now, description="时间戳")
